src/cargo/ops/cargo_add/mod.rs

```diff
@@ -314,7 +314,7 @@ fn resolve_dependency(
         } else {
             let source = crate::sources::PathSource::new(&path, src.source_id()?, config);
             let packages = source.read_packages()?;
-            let package = infer_package(packages, &src)?;
+            let package = infer_package(packages, &src, false)?;
```

Comment:
The intention from the `bool` parameter isn't obvious from the function call, so I think it'd be better if we did this a different way.

Reply:
Please check out https://github.com/rust-lang/cargo/pull/11186/commits/a62b1ada499df523931f06ccff7f0d5f62b5841f